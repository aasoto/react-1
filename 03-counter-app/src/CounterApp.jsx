import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => {

  const [ counter, setCounter ] = useState( value );

  const handleAdd = () => {
    setCounter((c) => {
      return c + 1;
    })
  }

  const handleSubtract = () => {
    setCounter((c) => {
      return c - 1;
    })
  }

  const handleReset = () => {
    setCounter(value)
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2 aria-label='counter'> { counter } </h2>

      <button aria-label='increment' onClick={ handleAdd }> +1 </button>
      <button aria-label='decrement' onClick={ handleSubtract }> -1 </button>
      <button aria-label='reset' onClick={ handleReset }> Reset </button>

    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}