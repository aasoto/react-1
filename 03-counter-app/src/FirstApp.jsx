const newMessage = {
  message: 'Bienvenidos a mi primera aplicación.',
  title: ' Andrés Soto React'
}

const getMessage = () => {
  return newMessage.message;
}

const getTitle = () => {
  return newMessage.title;
}

const sumar = (a, b) => {
  return a + b;
}

export const FirstApp = () => {
  /**
   * Para agrupar un numero de elementos debe haber un nodo padre,
   * el cual se puede definir con la etiqueta <Fragment></Fragment>
   * de manera resumida <></>
   */
  return (
    <>
      <h1>{getTitle()}</h1>
      <h3>{sumar(5, 3)}</h3>
    </>
  );
}