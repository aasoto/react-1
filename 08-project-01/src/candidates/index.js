export * from './data';
export * from './pages';
export * from './helpers';
export * from './components';
export * from './context';