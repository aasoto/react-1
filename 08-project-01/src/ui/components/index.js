export * from './Navbar';
export * from './OptionsNavbar';
export * from './Sidebar';