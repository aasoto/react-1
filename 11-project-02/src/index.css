@tailwind base;
@tailwind components;
@tailwind utilities;

.navbar {
  @apply fixed z-20 flex justify-center items-center w-full h-24 p-4
}

.navbar nav{
  @apply w-full h-full backdrop-blur-md bg-white/50 rounded-lg px-5 md:px-10 py-2 shadow-lg
}

.navbar .nav-container{
  @apply flex justify-between items-center
}

.navbar .nav-section{
  @apply flex items-center gap-5
}

.navbar h1{
  @apply text-black font-bold text-2xl italic
}