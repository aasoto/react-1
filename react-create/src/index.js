// Variables y constantes

const nombre = 'AndrÃ©s';
let