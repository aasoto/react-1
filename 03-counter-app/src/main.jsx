import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import HelloWorldApp from './HelloWorld';
import { PropsApp } from './PropsApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <PropsApp title="Probando PropTypes" subTitle='Esto es un subtÃ­tulo'/>
    {/* <CounterApp value={4} /> */}
  </React.StrictMode>
);