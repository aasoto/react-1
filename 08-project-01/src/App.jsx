import { AuthProvider } from "./auth"
import { Navbar, Sidebar, SidebarProvider } from "./ui"

export const App = () => {
  return (
    <AuthProvider>
      <SidebarProvider>
        <Navbar />
        <Sidebar />
      </SidebarProvider>
      {/* <div className="bg-slate-800 w-full h-screen">
        <img src="images/miss-universe-logo.svg" alt="miss-universe-logo" />
        <h1 className="text-3xl text-white font-bold underline">
          Hello world!
        </h1>
      </div> */}
    </AuthProvider>
  )
}
