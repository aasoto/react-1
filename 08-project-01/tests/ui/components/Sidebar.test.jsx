import { render } from "@testing-library/react";
import { Sidebar, SidebarContext } from "../../../src/ui";

const showSidebarMock = jest.fn();

describe('Pruebas de <Sidebar />', () => {

  test('debe hacer match con el snapshot', () => {

    const {container} = render(
      <SidebarContext.Provider value={{showSidebar: showSidebarMock}}>
        <Sidebar/>
      </SidebarContext.Provider>
    );

    expect(container).toMatchSnapshot();

  });

});