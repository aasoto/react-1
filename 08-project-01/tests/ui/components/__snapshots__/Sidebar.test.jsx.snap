// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <Sidebar /> debe hacer match con el snapshot 1`] = `
<div>
  <aside
    class="fixed z-10 top-24 left-5 h-5/6 w-72 p-3 shadow-xl bg-white rounded-lg animate__animated block animate__fadeInLeft"
  >
    <div
      class="flex flex-col gap-5"
    >
      <input
        class="w-full h-12 px-5 border rounded-md border-gray-400 placeholder:text-gray-400"
        name="searchingBox"
        placeholder="Type the name of the contestant"
        type="text"
        value=""
      />
    </div>
  </aside>
</div>
`;
